lib/spack/spack/compiler.py

```diff
@@ -174,6 +176,111 @@ def check(exe):
     def version(self):
         return self.spec.version
 
+    @property
+    def verbose_flag(self):
+        """
+        This property should be overridden in the compiler subclass if a
+        verbose flag is available.
+
+        If it is not overridden, it is assumed to not be supported.
```

Comment:
This isn't actually true -- if this is not overridden, it looks like `_determine_implicit_link_paths` still uses the result.  I think it should just skip detection.

Reply:
In `implicit_link_paths`, the cache is set if the flag isn't set otherwise `_determine_implicit_link_paths` is called to fill the cache.